pyproject.toml

```diff
@@ -1,51 +1,95 @@
+# License: MIT
+# Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
 [build-system]
-requires = ["setuptools == 65.3.0", "setuptools_scm[toml] == 7.0.5", "wheel"]
+requires = [
+  "setuptools == 67.7.2",
+  "setuptools_scm[toml] == 7.1.0",
+  "frequenz-repo-config[lib] == 0.3.0",
+]
 build-backend = "setuptools.build_meta"
 
-
 [project]
 name = "frequenz-channels"
 description = "Channel implementations for Python"
 readme = "README.md"
 license = { text = "MIT" }
-keywords = ["frequenz", "channel"]
+keywords = ["frequenz", "python", "lib", "channels", "channel"]
 classifiers = [
   "Development Status :: 3 - Alpha",
   "Intended Audience :: Developers",
   "License :: OSI Approved :: MIT License",
   "Programming Language :: Python :: 3",
   "Programming Language :: Python :: 3 :: Only",
   "Topic :: Software Development :: Libraries",
+  "Typing :: Typed",
 ]
 requires-python = ">= 3.11, < 4"
-dependencies = ["watchfiles >= 0.15.0, < 0.20.0"]
+dependencies = [
+  "typing-extensions >= 4.5.0, < 5",
+  "watchfiles >= 0.15.0, < 0.20.0",
+]
 dynamic = ["version"]
 
 [[project.authors]]
 name = "Frequenz Energy-as-a-Service GmbH"
 email = "floss@frequenz.com"
 
 [project.optional-dependencies]
-docs = [
+dev-docstrings = [
+  "pydocstyle == 6.3.0",
+  "darglint == 1.8.1",
+  "tomli == 2.0.1",      # Needed by pydocstyle to read pyproject.toml
+]
+dev-formatting = ["black == 23.3.0", "isort == 5.12.0"]
+dev-mkdocs = [
   "mike == 1.1.2",
   "mkdocs-gen-files == 0.5.0",
   "mkdocs-literate-nav == 0.6.0",
-  "mkdocs-material == 9.1.17",
+  "mkdocs-material == 9.1.16",
```

Comment:
why going backwards?

Reply:
Oversight, the templates are outdated. I think it will be very annoying to keep templates up to date, so I thought it was better to just leave the projects to get updated via dependabot. But of course when migrating/updating a project we shouldn't revert dependencies, so I will fix, thanks!